### ISSUE TITLE
Question about list[A | B | C]

### ISSUE THREAD

**Issue by s-banach** [microsoft/pyright#5063]
Good morning,
I know there have been issues about this before, I'd just like to clarify.

I might naively type a function like
```python
def f(x: list[A | B | C]):
    ...
```
with the intention that users of my package could pass in `list[A]` if they wanted.
But they'll get a type error if they do this.  
The justification for treating this as an error is that lists are mutable.  
If the type checker treated `list[A]` as an instance of `list[A|B|C]`,
then the function `f` might modify `x` by adding elements of type `B` or `C` to it.

(1) Is the only correct annotation for `x` to include all 7 combinations of `A,B,C`?

(2) Is there any way to annotate `x` as immutable, so that users can pass in `list[A]` and the type checker can understand that it's safe to treat `list[A]` as `list[A|B|C]`?
If not, have there been any python PEPs to add immutable lists?

Thanks for your time.

**Comment by erictraut** [comment]
Converting to discussion since this isn't a bug report.
